### ISSUE TITLE
False positive when closing a named coproc's fd

### ISSUE THREAD

**Issue by silicon-salamander** [koalaman/shellcheck#3130]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2093, SC1083, SC1083
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

coproc tr { tr a b; }
echo aaa >&"${tr[1]}"

exec {tr[1]}>&- # errors here

cat <&"${tr[0]}"
```

#### Here's what shellcheck currently says:
```
Line 6	SC2093: Remove "exec " if script should continue after this command.
Line 6	SC1083: This { is literal. Check expression (missing ;/\n?) or quote it.
Line 6	SC1083: This } is literal. Check expression (missing ;/\n?) or quote it.
```
#### Here's what I wanted or expected to see:

Expected no errors. `exec {tr[1]}>&-` works correctly on bash version 5.2.37.
According to this SO answer this syntax is not valid in bash versions prior to 4.3.
[https://unix.stackexchange.com/a/86372](https://unix.stackexchange.com/a/86372)

Also relevant bash change log entry.
[https://git.savannah.gnu.org/cgit/bash.git/tree/CWRU/changelog?id=30a978b7d808c067219c95be88c4979b6a7aa251#n3427](https://git.savannah.gnu.org/cgit/bash.git/tree/CWRU/changelog?id=30a978b7d808c067219c95be88c4979b6a7aa251#n3427)
